UnitTests/GitCommandsTests/FileAssociatedIconProviderTests.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System.IO.Abstractions;
+using FluentAssertions;
+using GitCommands;
+using NSubstitute;
+using NUnit.Framework;
+
+namespace GitCommandsTests
+{
+    [TestFixture]
+    public class FileAssociatedIconProviderTests
+    {
+        private FileBase _file;
+        private IFileSystem _fileSystem;
+        private FileAssociatedIconProvider _iconProvider;
+
+
+        [SetUp]
+        public void Setup()
+        {
+            _file = Substitute.For<FileBase>();
+            _fileSystem = Substitute.For<IFileSystem>();
+            _fileSystem.File.Returns(_file);
+            _iconProvider = new FileAssociatedIconProvider(_fileSystem);
+        }
+
+
+        [TestCase(null)]
+        [TestCase("")]
+        [TestCase("\t")]
+        public void Get_should_return_null_if_path_null_or_empty(string fullPath)
+        {
+            _iconProvider.Get(fullPath).Should().BeNull();
+        }
+
+        [Test]
+        public void Get_should_return_null_if_file_not_exist()
```

Comment:
Do we need that assumption? For old revisions there may be shown files that no longer exist in the working dir, but we still can show them with the icon associated to its extension.